### ISSUE TITLE
Can't cancel autoplay when pauseOnHover is true

### ISSUE THREAD

**Issue by thomashigginbotham** [kenwheeler/slick#731]
If you try to cancel autoplay by calling slickPause(), the autoplay will resume when the mouseleave event fires. If you try to cancel autoplay with slickSetOption('autoplay', false), there appears to be no effect at all.

I had a need to stop autoplay after a user starts playing a video in one of the slides, and the only way to do it was to call slickPause() and then remove the mouseleave event listener to prevent autoplay from resuming again.

See http://jsfiddle.net/jyccpynw/ for an example.

**Comment by kenwheeler** [comment]
After reviewing, this is really just conflicting settings. It is probably best handled on a case per case basis, rather than rigged into core.

**Comment by thomashigginbotham** [comment]
So how do you cancel autoplay after it has started? slickSetOption('autoplay', false) doesn't work, and slickPause() only works until the mouse cursor leaves the carousel.

**Comment by nominalaeon** [comment]
I'm having this same issue. OnSlideHover shouldn't pause autoplay, but dot-hover (or at least dot-click) should stop it entirely.

I don't want the autoplay to resume if a user has clicked a dot. Even slickPause() gets over-ruled, so I can't spin up my own bound event.

I need an all-powerful way to stop autoplay and have it never return.

**Comment by joxxoxo** [comment]
Same for me. Is there a way to stop autoplay entirely? I need to do it if edge reached or any dot clicked

**Comment by nominalaeon** [comment]
Sorry, I meant to double back and write my solution. I wound up messaging kWheels and he walked me through it:

    bindEvents: function () {
      this.$navDots = this.$root.find('.slick-dots').find('button');
      this.$navDots.on('click', $.proxy(this.onDotClick, this));
    },
    onDotClick: function () {
      this.$slickFrame.slick('slickPause');
      this.$slickFrame.slick('setOption', 'autoplay', false, false);
    }

Basically you just pause the player then change its autoplay option. I forget what the double "false, false" part does.

**Comment by joxxoxo** [comment]
@nominalaeon thanks for the code. Second false is `refresh`: `Set refresh to true if it's a UI update.`
If anyone is interested I'm thinking about this solution:

    var stopSlider = function() {
      slick = homeSlider.slick('getSlick');
      slick.slickSetOption("autoplay", false, false);
      slick.autoPlay = $.noop;
    };
    homeSlider.on('afterChange', function(_event, slick, currentSlideIndex){
      // Stop if showing last slide
      if((slick.$slides.length - 1) == currentSlideIndex) { stopSlider(); }
    });
    homeSlider.slick({
      fade: true,
      autoplay: true,
      speed: 1500,
      autoplaySpeed: 4500,
      dots: true,
      infinite: false
    });
    homeSlider.slick('getSlick').$dots.on('click', 'li', function() { stopSlider(); });

**Comment by nominalaeon** [comment]
Nice, did you try it yet? Let me know if it works without the slickPause, I'll be able to save myself an extra line of code!

Also, I think you can remove the anonymous function and call

    homeSlider.slick('getSlick').$dots.on('click', 'li', stopSlider);
